src/decorators/is-date.spec.ts

```diff
@@ -29,12 +34,39 @@ describe('IsDate', () => {
         Result.err('date is not formatted as `yyyy-mm-dd`')
       );
     });
+
+    it('transforms to date formatted string', async () => {
+      const dateToUse = new Date();
+      const obj = new TestSerialize();
+      obj.date = dateToUse;
+
+      expect(await output(obj)).toStrictEqual({ date: dateToUse.toISOString().slice(0, 10) });
+    });
+
+    it('pass through when field is undefined', async () => {
+      const obj = new TestSerialize();
+
+      expect(await output(obj)).toStrictEqual({});
+    });
+
+    it('pass through value if no serialize', async () => {
```

Comment:
if the 'format' is specified, i think class should not accept plain Date objects